app/services/integrations/xero/create_service.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Integrations
+  module Xero
+    class CreateService < BaseService
+      def call(**args)
+        organization = Organization.find_by(id: args[:organization_id])
+
+        unless organization.premium_integrations.include?('xero')
+          return result.not_allowed_failure!(code: 'premium_integration_missing')
+        end
+
+        integration = Integrations::XeroIntegration.new(
+          organization:,
+          name: args[:name],
+          code: args[:code],
+          connection_id: args[:connection_id]
+        )
+
+        integration.save!
+
+        if integration.type == 'Integrations::XeroIntegration'
```

Comment:
To clarify, `integration` is created on line 13 using `integration = Integrations::XeroIntegration.new(`, not sure how the type could be something other than `Integrations::XeroIntegration`